reporting/server/session.h

```diff
@@ -93,21 +104,54 @@ class session : public std::enable_shared_from_this<session>
     boost::beast::websocket::stream<boost::beast::tcp_stream> ws_;
     boost::beast::flat_buffer buffer_;
     std::string response_;
-    BackendInterface const& backend_;
-    SubscriptionManager& subscriptions_;
+
+    std::shared_ptr<BackendInterface> backend_;
+    std::weak_ptr<SubscriptionManager> subscriptions_;
```

Comment:
I _think_ that `session::subscribers_` needs to be a `std::weak_ptr` to prevent a cyclic dependency when between SubscriptionManager and session.

We pass `subscribers_.lock()` into `buildResponse()`, so at any point during the handler the SubscriptionManager should exist.

Reply:
You are correct that this needs to be a weak ptr to prevent a cyclic dependency